access/pagerduty/app.go

```diff
@@ -288,12 +280,64 @@ func (a *App) onDeletedRequest(ctx context.Context, req access.Request) error {
 		return trace.Wrap(err)
 	}
 
-	if err := a.bot.ResolveIncident(ctx, reqID, pluginData.PagerdutyData, "expired"); err != nil {
+	if err := a.bot.ResolveIncident(ctx, reqID, pluginData.PagerdutyData.ID, "expired"); err != nil {
 		return trace.Wrap(err)
 	}
 
-	log.WithField("request_id", reqID).Info("Successfully marked request as expired")
+	log.Info("Successfully marked request as expired")
+
+	return nil
+}
+
+func (a *App) setRequestState(ctx context.Context, reqID, incidentID, userEmail string, state access.State) error {
+	log := utils.GetLogger(ctx)
+	var resolution string
+
+	switch state {
+	case access.StateApproved:
+		resolution = "approved"
+	case access.StateDenied:
+		resolution = "denied"
+	}
```

Comment:
Best have a `default` case here.  Either return an `error` indicating an unexpected input, or fallback to `state.String()`.